File: Java/src/main/java/br/com/locaweb/locamail/api/repository/RespostaEmailRepository.java
package br.com.locaweb.locamail.api.repository;

import br.com.locaweb.locamail.api.model.Convidado;
import br.com.locaweb.locamail.api.model.RespostaEmail;
import jakarta.transaction.Transactional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;


@Repository
public interface RespostaEmailRepository extends JpaRepository<RespostaEmail, Long> {

    @Query(value = "SELECT * FROM T_LCW_RESPOSTA_EMAIL  WHERE id_email = :id_email", nativeQuery = true)
    public List<RespostaEmail> listarRespostasEmailPorIdEmail(@Param("id_email") Long id_email);

    @Query(value = "SELECT * FROM T_LCW_RESPOSTA_EMAIL  WHERE id_resposta_email = :id_resposta_email", nativeQuery = true)
    public RespostaEmail listarRespostaEmailPorIdRespostaEmail(@Param("id_resposta_email") Long id_resposta_email);

    @Transactional
    @Modifying
    @Query(value = "DELETE FROM T_LCW_RESPOSTA_EMAIL WHERE id_email = :id_email", nativeQuery = true)
    public void excluirRespostaEmailPorIdEmail(@Param("id_email") Long id_email);

}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/database/repository/RespostaEmailRepository.kt
package br.com.fiap.locawebmailapp.database.repository

import android.content.Context
import br.com.fiap.locawebmailapp.database.dao.InstanceDatabase
import br.com.fiap.locawebmailapp.model.RespostaEmail


class RespostaEmailRepository(context: Context) {

    private val respostaEmailDao =
        InstanceDatabase.getDatabase(context = context).respostaEmailDao()

    fun criarRespostaEmail(respostaEmail: RespostaEmail): Long {

        return respostaEmailDao.criarRespostaEmail(respostaEmail)

    }

    fun listarRespostaEmailPorIdRespostaEmail(id_resposta_email: Long): RespostaEmail {
        return respostaEmailDao.listarRespostaEmailPorIdRespostaEmail(id_resposta_email)
    }

    fun atualizarRespostaEmail(respostaEmail: RespostaEmail) {

        return respostaEmailDao.atualizarRespostaEmail(respostaEmail)

    }

    fun excluirRespostaEmail(respostaEmail: RespostaEmail) {

        return respostaEmailDao.excluirRespostaEmail(respostaEmail)

    }

    fun listarRespostasEmailPorIdEmail(id_email: Long): List<RespostaEmail> {

        return respostaEmailDao.listarRespostasEmailPorIdEmail(id_email)

    }

    fun excluirRespostaEmailPorIdEmail(id_email: Long) {
        return respostaEmailDao.excluirRespostaEmailPorIdEmail(id_email)
    }
}